pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -1397,27 +1397,48 @@ private static int integerPart(String dirtyString) {
     return Integer.parseInt(dirtyString.substring(start, end));
   }
 
+  // Validation check for integer values that need to be positive
+  // mainly added to check resultSetType, resultSetConcurrency and resultSetHoldability
+  private static boolean isPositive(int value) {
+    if (value < 0) {
+      return false;
+    }
+    return true;
+  }
+
   @Override
   public Statement createStatement(int resultSetType, int resultSetConcurrency,
       int resultSetHoldability) throws SQLException {
     checkClosed();
-    return new PgStatement(this, resultSetType, resultSetConcurrency, resultSetHoldability);
+    if (isPositive(resultSetType) && isPositive(resultSetConcurrency) && isPositive(resultSetHoldability)) {
```

Comment:
Why checking for "positive" property only?
Should we rather verify the supported values instead?

Reply:
Okay, makes sense. So validation for resultSetType, resultSetConcurrency, resultSetConcurrency, resultSetHoldability can be separated into different validation methods. Just wanted to double check if these the values that need to be validated for every param?

resultSetType 
- int TYPE_FORWARD_ONLY = 1003;
- int TYPE_SCROLL_INSENSITIVE = 1004;
- int TYPE_SCROLL_SENSITIVE = 1005;

resultSetConcurrency
- int CONCUR_READ_ONLY = 1007;
- int CONCUR_UPDATABLE = 1008;

resultSetHoldability
- int HOLD_CURSORS_OVER_COMMIT = 1;
- int CLOSE_CURSORS_AT_COMMIT = 2;